plugins/adr/src/alpha.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  createPlugin,
+  createSchemaFromZod,
+} from '@backstage/frontend-plugin-api';
+import { createSearchResultListItemExtension } from '@backstage/plugin-search-react/alpha';
+import { AdrDocument } from '@backstage/plugin-adr-common';
+
+export * from './translations';
+
+function isAdrDocument(result: any): result is AdrDocument {
+  return result.entityRef;
+}
+
+/**
+ * React extension used to render results on Search page or modal
+ * @alpha
+ */
+export const AdrSearchResultListItemExtension =
+  createSearchResultListItemExtension({
+    id: 'adr',
+    at: 'plugin.search.page/items',
+    configSchema: createSchemaFromZod(z =>
+      z.object({
+        // TODO: Define how the icon can be configurable
```

Comment:
Do you happen to have any ideas, @Rugvip?

Reply:
Could be an input, could be an API, could be something else. I think it's fine to leave for now :p